quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -0,0 +1,1352 @@
+        let contents: Vec<u8>;
+
+        {
+            let mut data = self.data.write().unwrap();
+
+            // Check for the existence of index.
+            if !data.contains_key(&index_uri) {
+                return Err(MetastoreErrorKind::IndexDoesNotExist
+                    .with_error(anyhow::anyhow!("Index does not exist: {:?}", &index_uri)));
+            }
+
+            let metadata_set = data.get_mut(&index_uri).unwrap();
+
+            // Check for the existence of split.
+            if !metadata_set.splits.contains_key(&split_id) {
+                return Err(MetastoreErrorKind::DoesNotExist
+                    .with_error(anyhow::anyhow!("Split ID does not exist: {:?}", &split_id)));
+            }
+
+            let split_metadata = metadata_set.splits.get_mut(&split_id).unwrap();
+
+            // Check the split state.
+            match split_metadata.split_state {
+                SplitState::Published => {
+                    // If the split is already published, this API call returns a success.
+                    return Ok(());
+                }
+                SplitState::Staged => {
+                    // Update the split state to `Published`.
+                    split_metadata.split_state = SplitState::Published;
+                }
+                _ => {
+                    return Err(MetastoreErrorKind::SplitIsNotStaged
+                        .with_error(anyhow::anyhow!("Split ID is not staged: {:?}", &split_id)));
+                }
+            }
+
+            // Serialize meta data.
+            contents = match serde_json::to_vec(&metadata_set) {
+                Ok(c) => c,
+                Err(e) => {
+                    return Err(MetastoreErrorKind::InvalidManifest
+                        .with_error(anyhow::anyhow!("Failed to serialize metadata set: {:?}", e)));
+                }
+            };
+        }
+
+        // Put data back into storage.
+        self.storage
+            .put(&Path::new(&index_uri), PutPayload::from(contents))
+            .await
+            .map_err(|e| {
+                MetastoreErrorKind::InternalError
+                    .with_error(anyhow::anyhow!("Failed to put metadata set: {:?}", e))
+            })?;
+
+        Ok(())
+    }
+
+    async fn list_splits(
+        &self,
+        index_uri: IndexUri,
+        state: SplitState,
+        time_range: Option<Range<u64>>,
+    ) -> MetastoreResult<Vec<SplitMetadata>> {
+        // list of split metadata.
+        let mut splits: Vec<SplitMetadata> = Vec::new();
+
+        let mut data = self.data.write().unwrap();
+
+        // Check for the existence of index.
+        if !data.contains_key(&index_uri) {
+            return Err(MetastoreErrorKind::IndexDoesNotExist
+                .with_error(anyhow::anyhow!("Index does not exist: {:?}", &index_uri)));
+        }
+
+        let metadata_set = data.get_mut(&index_uri).unwrap();
+
+        // filter by split state.
+        let it = metadata_set
```

Comment:
`it` is not a good variable name as it does not help the reader.

`split_with_meta_matching_state_it` would be better.

Reply:
Changed.